query/common.d.ts

```diff
@@ -269,6 +277,46 @@ export interface Bindings extends Iterable<[RDF.Variable, RDF.Term]> {
   ) => Bindings;
 }
 
+/**
+ * A Path represents the sequence of Bindings or "hops" that connect one node to another in a graph.
+ * Each Bindings object contains the subject - predicate - object triple that is one hop in the path.
+ * 
+ * A Path instance is created using the PathFactory.
+ * 
+ */
+export interface Path extends Iterable<Bindings> {
+  type: 'path';
+  /**
+   * The number of Bindings in this path.
+   */
+  length: number;
+  /**
+   * Check if the path contains the given Variable-Term pair.
+   * @param key A variable-term pair.
+   */
+  has: (key: [RDF.Variable, RDF.Term]) => boolean;
```

Comment:
One last thing! 😅 I don't think this method makes sense for this interface, unless it means that this exists just somewhere along the path. But probably doesn't make sense.

Reply:
Yes, it just checks if the path already includes a node, i.e., ['s', <Alice>]. It could be useful in the future. I can comment it out for now.